bot/exts/evergreen/game.py

```diff
@@ -135,8 +151,47 @@ def __init__(self, bot: Bot):
         self.http_session: ClientSession = bot.http_session
 
         self.genres: Dict[str, int] = {}
-
-        self.refresh_genres_task.start()
+        self.headers = BASE_HEADERS
+
+        self.bot.loop.create_task(self.renew_access_token())
+
+    async def renew_access_token(self) -> None:
+        """Refeshes V4 access token a number of seconds before expiry. See `ACCESS_TOKEN_RENEWAL_WINDOW`."""
+        while True:
+            async with self.http_session.post(OAUTH_URL, params=OAUTH_PARAMS) as resp:
```

Comment:
I am wondering if its worth pulling this out into another function for ease of parsing the code?

Reply:
This does one thing and should be in one function.